webrtc/src/benchmarkRtc.ts

```diff
@@ -0,0 +1,168 @@
+import { chromium, Browser } from "playwright";
+import { Client } from "./client";
+import { Args } from "./types";
+import { getEncodingsReport, onEncodingsUpdate } from "./encodingReporter";
+
+const frontendAddress = "http://localhost:5005";
+const fakeVideo = "media/sample_video.mjpeg";
+const fakeAudio = "media/sample_audio.wav";
+const ENCODING_REPORT_PERIOD = 5;
+const INBOUD_TRACK_BANDWIDTH = 0.15 + 0.5 + 1.5;
+const OUTBOUND_TRACK_BANDWIDTH = 1.5;
+
+const delay = (s: number) => {
+  return new Promise((resolve) => setTimeout(resolve, 1000 * s));
+};
+
+export const runBenchmark = async (args: Args) => {
+  const client = new Client(args);
+  await client.purge();
+
+  const browsers = await addPeers(args);
+
+  console.log("Started all browsers, running benchmark");
+
+  const encodingReports = [];
+
+  let time = 0,
+    step = ENCODING_REPORT_PERIOD;
+  while (true) {
+    await delay(step);
+    step = Math.min(step, args.duration - time);
+    time += step;
+
+    const report = getEncodingsReport();
+    encodingReports.push({ ...report.toJson(), timestamp: time });
+
+    writeInPlace(
+      `${time} / ${args.duration}s (${Math.floor(
+        time / args.duration,
+      )}), Encodings: ${report.toString()}}`,
+    );
+
+    if (time == args.duration) break;
+  }
+
+  await cleanup(client, browsers);
+
+  console.log("\nBenchmark finished, closing");
+  process.exit(0);
+};
+
+const addPeers = async (args: Args) => {
+  const client = new Client(args);
+
+  let peersAdded = 0;
+  let peersInCurrentBrowser = 0;
+  let browsers: Array<Browser> = [];
+  let currentBrowser = await spawnBrowser(args.chromeExecutable);
+
+  while (peersAdded < args.peers) {
+    const roomId = await client.createRoom();
+
+    for (let j = 0; j < args.peersPerRoom && peersAdded < args.peers; j++) {
+      await startPeer({
+        browser: currentBrowser!,
+        client: client,
+        roomId: roomId,
+        active: j < args.activePeers,
+      });
+      peersAdded++, peersInCurrentBrowser++;
+
+      const { incoming, outgoing } = getTrackNumber(args);
+      writeInPlace(
+        `Browsers launched: ${peersAdded} / ${
+          args.peers
+        }  Expected network usage: Incoming ${
+          incoming * INBOUD_TRACK_BANDWIDTH
+        } Mbit/s, Outgoing ${outgoing * OUTBOUND_TRACK_BANDWIDTH} Mbit/s`,
+      );
+      await delay(args.peerDelay);
+
+      if (peersInCurrentBrowser == args.peersPerBrowser) {
+        browsers.push(currentBrowser!);
+        currentBrowser = await spawnBrowser(args.chromeExecutable);
+        peersInCurrentBrowser = 0;
+      }
+    }
+  }
+  console.log("");
```

Comment:
Do we need this?

Reply:
The previous print doesn't have endline.